machine_learning/data_downloader.py

```diff
@@ -0,0 +1,160 @@
+        ids = {'site_id': site_id, 'page_id': page_id,
+               'from_version_id': from_version_id, 'to_version_id': to_version_id}
+        return ids
+    except:
+        warnings.warn('Url "' + url + '" is not in the required format and no data will be downloaded for this row', Warning)
+
+def get_storage_uris(url):
+    """
+    Get uris of the versions stored in our db
+    Uses get_version_by_versionista_id from web_monitoring.db
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    from_version_uri, to_version_uri : string
+        To and from version uris to access content in db
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    ids = parse_url(url)
+    from_version_id = ids['from_version_id']
+    to_version_id = ids['to_version_id']
+    print(ids)
+    client = Client.from_env()
+    """
+    Some versionista uris have the format https://versionista.com/site_id/page_id/to_version_id:0/
+    which automatically directs to the diff with the previous version.
+    However, versions in our db are saved with their true version ids and when the from_version_id is '0',
+    this is handled by taking advantage of the optional parameters of get_version_uri.
+    """
+    if from_version_id == '0':
+        get_to_version = client.get_version_by_versionista_id(versionista_id=to_version_id)
+        to_version_uri = get_to_version['data']['uri']
+        valid_from_version_id = get_to_version['data']['source_metadata']['diff_with_previous_url'].split(':')[-1]
+        from_version_uri = client.get_version_by_versionista_id(versionista_id=valid_from_version_id)['data']['uri']
+        ids['from_version_id'] = valid_from_version_id
+    else :
+        to_version_uri = client.get_version_by_versionista_id(versionista_id=to_version_id)['data']['uri']
+        from_version_uri = client.get_version_uri(versionista_id=from_version_id)['data']['uri']
+    print(from_version_uri, to_version_uri)
+    return from_version_uri, to_version_uri, ids
+
+def download(filename, dirname):
+    """
+    Downloader to read csv and download content for the dataset
+
+    Parameters
+    ----------
+    filename : string
+        The csv file which has the urls in the 'Last Two - Side by Side' column
+    dirname : string
+        The directory name in which you want to download the content
+    """
+    path = os.getcwd()
+    if (not os.path.exists(dirname)):
+        os.mkdir(dirname)
+
+    download_log_handler = open(str(dirname) + '.txt', 'w')
+    filename = os.path.join(path, filename)
+    df = pd.read_csv(filename, header=0, error_bad_lines=False)
+    new_dir = os.path.join(path, dirname)
+    os.chdir(new_dir)
+    download_log_handler.write(time.asctime(time.localtime(time.time())) + '\n')
+    urls = df['Last Two - Side by Side']
+
+    for index in range(0, len(df['Last Two - Side by Side'])):
+        try:
```

Comment:
I agree with this. This is a good use of try/except because `download` is a batch operation. I would however try to provide more information about the failure, like this:

```python
try:
   ...
except Exception as exc:
   # include repr(exc) in the log message
```